File: check_tovec.m
f4m = importdata("tovErr_rk4m_fixed.log");
f5m = importdata("tovErr_rk5m_fixed.log");
f5f = importdata("tovErr_rk5f_fixed.log");
f5dp = importdata("tovErr_rk5dp_fixed.log");
f5ck = importdata("tovErr_rk5ck_fixed.log");

clr=[0.94118 0.50196 0.50196;0.56078 0.73725 0.56078;0.39216 0.58431 0.92941;0.93 0.57 0.13];

hold on;
p4m=plot(f4m(:,2),"k");
p5m=plot(f5m(:,2),"Color",clr(1,:));
p5f=plot(f5f(:,2),"Color",clr(2,:));
p5dp=plot(f5dp(:,2),"Color",clr(3,:));
p5ck=plot(f5ck(:,2),"Color",clr(4,:));

set(gca, "YScale", "log", "Box", "on");
legend([p4m, p5m, p5f, p5dp, p5ck], {'RK-4M','RK-5M','RK-5F','RK-5DP','RK-5CK'});legend('boxoff');

File: check_tovec.py
import numpy as np
import matplotlib.pyplot as plt

f4m = np.loadtxt("tovErr_rk4m_fixed.log", delimiter=',')
f5m = np.loadtxt("tovErr_rk5m_fixed.log", delimiter=',')
f5f = np.loadtxt("tovErr_rk5f_fixed.log", delimiter=',')
f5dp = np.loadtxt("tovErr_rk5dp_fixed.log", delimiter=',')
f5ck = np.loadtxt("tovErr_rk5ck_fixed.log", delimiter=',')

plt.figure()
ax=plt.gca()

ax.plot(f5ck[:,0], f5ck[:,1], "-", label = "Cash-Karp", color=plt.cm.Set2(0), linewidth=1, alpha=0.5)
ax.plot(f5f[:,0], f5f[:,1], "-", label = "RKF-5 (4)", color=plt.cm.Set2(1), linewidth=1, alpha=0.5)
ax.plot(f5m[:,0], f5m[:,1], "-", label = "Dormand-Prince (M)", color=plt.cm.Set2(2), linewidth=1, alpha=0.5)
ax.plot(f5dp[:,0], f5dp[:,1], "-", label = "Dormand-Prince", color=plt.cm.Set2(3), linewidth=1, alpha=0.5)
ax.plot(f4m[:,0], f4m[:,1], "k-", linewidth=0.5, label = "RK-4 (3)", alpha=0.5)

#ax.plot(np.array([0,11.5]), np.array([1e-8,1e-8]), "k--", linewidth = 2)
#ax.plot(np.array([0,11.5]), np.array([1e-10,1e-10]), "k--", linewidth = 2)

ax.minorticks_on()
ax.tick_params(which="both", top=1, right=1)
ax.tick_params(which="both", axis="both", direction="in", width=2.0)
ax.tick_params(which="major", length=6.0)
ax.tick_params(which="minor", length=4.0)

ax.spines["top"].set_linewidth(2)
ax.spines["right"].set_linewidth(2)
ax.spines["bottom"].set_linewidth(2)
ax.spines["left"].set_linewidth(2)

ax.set_yscale("log")
ax.set_ylim(1e-13,1e-4)

labels = ax.get_xticklabels()+ax.get_yticklabels()
for label in labels:
    label.set_fontweight("bold")

ax.set_xlabel("Radius [$\mathrm{km}$]", fontdict={"weight":"bold","size":12})
ax.set_ylabel("Local Error", fontdict={"weight":"bold","size":12})

ax.legend(loc="upper center", frameon=0, fontsize=9.6, borderpad=1.2, ncol=1)
#plt.savefig("err_fix_h1.pdf")
plt.show()